shell/common/shell_test_platform_view.cc

```diff
@@ -57,5 +59,33 @@ std::unique_ptr<ShellTestPlatformView> ShellTestPlatformView::Create(
   }
 }
 
+ShellTestPlatformViewBuilder::ShellTestPlatformViewBuilder(Config config)
+    : config_(config) {}
+
+std::unique_ptr<PlatformView> ShellTestPlatformViewBuilder::operator()(
```

Comment:
nit: avoid defining a custom operator for this, just make a `Build` method.

Reply:
The custom operator is to make this class assignable to the current `std::function` parameter. We can turn this into a `Build` method instead, but then the several cases that currently define a custom closure will have to define a new class that accepts a closure. Should I do it?

Reply:
```cpp
  Shell::CreateCallback<PlatformView> platform_view_create_callback =
      [task_runners, main_context](flutter::Shell& shell) {
        auto result = std::make_unique<TestPlatformView>(shell, task_runners);
        ON_CALL(*result, CreateRenderingSurface())
            .WillByDefault(::testing::Invoke([main_context] {
              auto surface = std::make_unique<MockSurface>();
              ON_CALL(*surface, GetContext())
                  .WillByDefault(Return(main_context.get()));
              ON_CALL(*surface, IsValid()).WillByDefault(Return(true));
              ON_CALL(*surface, MakeRenderContextCurrent())
                  .WillByDefault(::testing::Invoke([] {
                    return std::make_unique<GLContextDefaultResult>(true);
                  }));
              return surface;
            }));
        return result;
      };

  auto shell = CreateShell({
      .settings = settings,
      .task_runners = task_runners,
      .platform_view_create_callback = platform_view_create_callback,
  });
```

https://github.com/flutter/engine/blob/6699bbb43b9d90d49d374ca215c52b383a77b7c1/shell/common/shell_unittests.cc#L1608

Reply:
That's fine then.